src/aelin-pool.ts

```diff
@@ -172,5 +174,47 @@ export function handleAcceptDeal(event: AcceptDealEvent): void {
   acceptDealEntity.aelinFee = event.params.aelinFee;
   acceptDealEntity.sponsorFee = event.params.sponsorFee;
 
+  let vestingDeal = VestingDeal.load(
+    event.params.purchaser.toHex() + "-" + event.params.dealAddress.toHex()
+  );
+  if (vestingDeal === null) {
+    let poolCreated = PoolCreated.load(event.address.toHex());
+    if (poolCreated === null) {
+      log.error("trying to find a pool not saved with address: {}", [
+        event.address.toHex(),
+      ]);
+      return;
+    }
+    let dealDetail = DealDetail.load(event.params.dealAddress.toHex());
+    if (dealDetail === null) {
+      log.error("trying to find a deal not saved with address: {}", [
+        event.params.dealAddress.toHex(),
+      ]);
+      return;
+    }
+
+    let exp = DEAL_WRAPPER_DECIMALS.minus(BigInt.fromI32(poolCreated.purchaseTokenDecimals))
+    // @ts-ignore 
+    let dealTokenAmount = event.params.poolTokenAmount.times(BigInt.fromI32(10).pow(<u8>exp.toI32()));

```

Comment:
as long as this works that is fine. I will try and do some testing on testnet against your subgraph as well